src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentStates.cs

```diff
@@ -69,8 +69,8 @@ public bool Contains(DocumentId id)
         public bool TryGetState(DocumentId documentId, [NotNullWhen(true)] out TState? state)
             => _map.TryGetValue(documentId, out state);
 
-        public TState? GetState(DocumentId documentId)
-            => _map.TryGetValue(documentId, out var state) ? state : null;
+        public TState? GetState(DocumentId? documentId)
+            => documentId != null && _map.TryGetValue(documentId, out var state) ? state : null;
```

Comment:
Ugh seeing the place we did this for ProjectId terrified me; can we do something else instead? What needed this?

Reply:
i will add this to the fallout list.